core/src/main/java/com/novoda/merlin/receiver/ConnectivityReceiver.java

```diff
@@ -23,28 +23,33 @@ public MerlinsBeard createMerlinsBeard(Context context) {
 
         MerlinBinderRetriever merlinBinderRetriever = new MerlinBinderRetriever() {
             @Override
-            public IBinder retrieveMerlinLocalServiceBinderIfAvailable(Context context) {
-                return peekService(context, new Intent(context, MerlinService.class));
+            public MerlinService.ConnectivityChangesNotifier retrieveConnectivityChangesNotifierIfAvailable(Context context) {
+                IBinder iBinder = peekService(context, new Intent(context, MerlinService.class));
+                if (iBinder != null
```

Comment:
It is also fine as it is but when you do `instanceof` check you don't need a null check beforehand, do you?